google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractStructReader.java

```diff
@@ -429,6 +437,18 @@ private void checkNonNullOfType(int columnIndex, Type expectedType, Object colum
     checkNonNull(columnIndex, columnNameForError);
   }
 
+  private void checkNonNullOfTypes(
+      int columnIndex, List<Type> expectedTypes, Object columnNameForError) {
+    Type actualType = getColumnType(columnIndex);
+    checkState(
+        expectedTypes.contains(actualType),
+        "Column %s is not of correct type: expected one of [%s] but was %s",
+        columnNameForError,
+        expectedTypes.stream().map(Type::toString).collect(Collectors.joining(", ")),
```

Comment:
would this be executed everytime we call checkNonNullOfTypes? If so, seems too expensive to me.

Reply:
Good point, I've updated it to just take a fixed string for now.